airflow/contrib/hooks/databricks_hook.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import requests
+
+from airflow import __version__
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+SUBMIT_RUN_ENDPOINT = ('POST', 'api/2.0/jobs/runs/submit')
+GET_RUN_ENDPOINT = ('GET', 'api/2.0/jobs/runs/get')
+USER_AGENT_HEADER = {'user-agent': 'airflow-{v}'.format(v = __version__)}
+
+class DatabricksHook(BaseHook):
+    """
+    Interact with Databricks.
+    """
+    def __init__(self, databricks_conn_id='databricks_default', timeout_seconds=60, retry_limit=3):
+        self.databricks_conn_id = databricks_conn_id
+        self.databricks_conn = self.get_connection(databricks_conn_id)
+        self.timeout_seconds = timeout_seconds
+        self.retry_limit = retry_limit
+
+    def _do_api_call(self, endpoint_info, api_parameters):
+        method, endpoint = endpoint_info
+        url = 'https://{host}/{endpoint}'.format(host=self.databricks_conn.host,
+                                                 endpoint=endpoint)
+        auth = (self.databricks_conn.login, self.databricks_conn.password)
+        if method == 'GET':
+            request_func = requests.get
+        elif method == 'POST':
+            request_func = requests.post
+        else:
+            raise AirflowException('Unexpected HTTP Method')
+
+        for _ in range(self.retry_limit):
+            attempt_num = _ + 1
+            try:
+                response = request_func(url,
+                                        json=api_parameters,
+                                        auth=auth,
+                                        headers=USER_AGENT_HEADER,
+                                        timeout=self.timeout_seconds,
+                                        verify=False)
+                if response.status_code == 404:
+                    # In this case, the user probably made a mistake.
```

Comment:
I would only return response.json if the return code is 200; otherwise, raise an exception